runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -582,9 +582,78 @@ impl wasmer_vm::Tunables for &Wasmer2VM {
         Ok(Arc::new(LinearTable::from_definition(&ty, &style, vm_definition_location)?))
     }
 
-    fn regular_op_cost(&self) -> u64 {
-        self.config.regular_op_cost.into()
+    fn stack_init_gas_cost(&self, stack_size: u64) -> u64 {
+        ((u64::from(self.config.regular_op_cost) + 7) / 8).saturating_mul(stack_size)
     }
+
+    /// Instrumentation configuration: stack limiter config
+    fn stack_limiter_cfg(&self) -> Box<dyn finite_wasm::max_stack::SizeConfig> {
+        Box::new(MaxStackCfg)
+    }
+
+    /// Instrumentation configuration: gas accounting config
+    fn gas_cfg(&self) -> Box<dyn finite_wasm::wasmparser::VisitOperator<Output = u64>> {
+        Box::new(GasCostCfg(u64::from(self.config.regular_op_cost)))
+    }
+}
+
+struct MaxStackCfg;
+
+impl finite_wasm::max_stack::SizeConfig for MaxStackCfg {
+    fn size_of_value(&self, ty: finite_wasm::wasmparser::ValType) -> u8 {
+        use finite_wasm::wasmparser::ValType;
+        match ty {
+            ValType::I32 => 4,
+            ValType::I64 => 8,
+            ValType::F32 => 4,
+            ValType::F64 => 8,
+            ValType::V128 => 16,
+            ValType::FuncRef => 8,
+            ValType::ExternRef => 8,
+        }
+    }
+    fn size_of_function_activation(&self, locals: &prefix_sum_vec::PrefixSumVec<finite_wasm::wasmparser::ValType, u32>) -> u64 {
+        let mut res = 0;
+        res += locals
+            .max_index()
+            .map(|l| u64::from(*l).saturating_add(1))
+            .unwrap_or(0)
+            * 8;
+        // TODO: make the above take into account the types of locals by adding an iter on PrefixSumVec that returns (count, type)
+        // THIS MUST HAPPEN BEFORE RELEASING THE PROTOCOL VERSION, SO PREFERABLY BEFORE LANDING
+        res += 32; // Rough accounting for rip, rbp and some registers spilled. Not exact.
```

Comment:
I’d say it would be prudent to reserve a little more space here (64 bytes?) in case we needed to start saving more stuff for whatever reason (e.g. due to platform differences if/when we port to aarch64 or similar)